auth/src/main/java/com/firebase/ui/auth/ui/idp/AuthMethodPickerActivity.java

```diff
@@ -185,9 +234,25 @@ public void onFailure(Bundle extra) {
     @Override
     public void onClick(View view) {
         if (view.getId() == R.id.email_provider) {
-            startActivityForResult(
-                    RegisterEmailActivity.createIntent(this, mActivityHelper.getFlowParams()),
-                    RC_EMAIL_FLOW);
+            if (mActivityHelper.getFlowParams().isReauth) {
+                String reauthEmail = mActivityHelper.getFlowParams().reauthEmail;
+                if (reauthEmail == null) {
+                    Log.e(TAG, "Attempting to reauth with email but no reauthEmail provided");
```

Comment:
As per Sam's comment reauthEmail is now removed, and the user is required to be logged in to start, so this whole block can go away